pkg/registrymirror/store.go

```diff
@@ -206,7 +206,7 @@ func (s *FileStore) PullSources(srcImage string) ([]string, error) {
 	}
 
 	if reg == nil {
-		return []string{srcImage}, nil
+		return nil, os.ErrNotExist
```

Comment:
curious why is this an error now? (future-proofing this comment: should we add a comment here explaining why?)

Reply:
It's an error now because:

- design wise it didn't make sense for the mirror store to be deciding what happens when no mirrors were found, clients were unaware nothing was found and could not react accordingly.
- the `srcImage` field (which for current usage is the output of `image.GetName().GetFullName()` ) does not / will not contain the 'id' of an image that was referenced by tag - and a client (in this case Sensor's enricher) did not know if the returned 'mirrors' were actually derived from the mirroring CRs or not - so the image ID was being lost.  

By returning an error the client can decide next steps (such as attempting a scan with the source image directly or perhaps bubbling the error)

Reply:
sounds good to me. Wondering if we should, instead, modify the check above and just return when `err != nil` (can do the `errors.Is` check in the client for `os.ErrNotExist`. Actually, according to the docs, we should use `fs.ErrNotExist`, instead - https://pkg.go.dev/os#IsNotExist

Reply:
Also, a few lines lower than this, I see `if ref == srcImage && reg.Blocked {`

why do we not just check `if reg.Blocked`? We still want to attempt, even if the mirror is blocked?

Also, how come we ignore the `Endpoint` field from the `PullSource`? I know out of scope for this PR, but I figure I should ask before I forget

Reply:
> sounds good to me. Wondering if we should, instead, modify the check above and just return when err != nil (can do the errors.Is check in the client for os.ErrNotExist. 

Yea, good call, given the changes just returning err should be OK.

> Actually, according to the docs, we should use fs.ErrNotExist, instead - https://pkg.go.dev/os#IsNotExist

Wonder if thats a typo hah. 

> Also, a few lines lower than this, I see if ref == srcImage && reg.Blocked {
> 
> why do we not just check if reg.Blocked? We still want to attempt, even if the mirror is blocked?

Blocked is set when the CR indicates `NeverContactSource`, if we just use `reg.Blocked` then no mirrors will be processed when  `NeverContactSource` is set - which is incorrect, we still want to attempt to contact the mirrors even when we should NOT contact the source.

> Also, how come we ignore the Endpoint field from the PullSource? I know out of scope for this PR, but I figure I should ask before I forget

Couldn't remember - ran a quick test and found Endpoint only references the 'registry' portion of the image (no repo, tag, digest, etc.) - we only care about the full image so we use the pull source's Reference instead.